### PR TITLE
feat: Windows images

### PR DESCRIPTION
**Pull request by lemeurherve**
Ref: https://github.com/jenkins-infra/helpdesk/issues/2873
Used https://github.com/jenkins-infra/pipeline-library/pull/372
Use https://github.com/jenkins-infra/pipeline-library/pull/385

### CODE REVIEW

File: maven/jdk8/Dockerfile.nanoserver [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Use https://github.com/jenkinsci/jnlp-agents/blob/994e06e18fa494ef6d7f70e97499542ebc8feef7/maven/jdk8/Dockerfile as base for now
```
Comment (lemeurherve) [inline maven/jdk8/Dockerfile.nanoserver:1]:
```suggestion
# Use https://github.com/jenkinsci/jnlp-agents as base for now
```

File: Jenkinsfile_k8s [hunk 2]
```diff
@@ -1,41 +1,14 @@
...
-    ])
-  },
-  'maven-jdk17': {
-     buildDockerAndPublishImage('inbound-agent-maven:jdk17', [
-      mainBranch: principalBranch,
-      dockerfile: 'maven/jdk17/Dockerfile',
+  'maven-jdk8-nanoserver': {
+     buildDockerAndPublishImage('inbound-agent-maven-nanoserver:jdk8', [
+      dockerfile: 'maven/jdk8/Dockerfile.nanoserver',
+      agentLabels: 'windows',
```
Comment (lemeurherve) [inline Jenkinsfile_k8s:10]:
```suggestion
	mainBranch: principalBranch,
        agentLabels: 'windows',
```

File: Jenkinsfile_k8s [hunk 3]
```diff
@@ -1,41 +1,15 @@
...
-  },
-  'maven-jdk11': {
-     buildDockerAndPublishImage('inbound-agent-maven:jdk11', [
-      mainBranch: principalBranch,
-      dockerfile: 'maven/jdk11/Dockerfile',
-    ])
-  },
-  'maven-jdk17': {
-     buildDockerAndPublishImage('inbound-agent-maven:jdk17', [
+  'maven-jdk8-nanoserver': {
```
Comment (lemeurherve) [inline Jenkinsfile_k8s:7]:
```suggestion
  'maven-jdk8': {
     buildDockerAndPublishImage('inbound-agent-maven:jdk8', [
      mainBranch: principalBranch,
      dockerfile: 'maven/jdk8/Dockerfile',
      useContainer: false,
    ])
  },
  'maven-jdk8-nanoserver': {
```

File: Jenkinsfile_k8s [hunk 4]
```diff
@@ -6,36 +8,15 @@ parallel(
...
+     buildDockerAndPublishImage('inbound-agent-maven-nanoserver:jdk8', [
```
Comment (lemeurherve) [inline Jenkinsfile_k8s:15]:
```suggestion
     buildDockerAndPublishImage('inbound-agent-maven:jdk8-nanoserver', [
```

File: Jenkinsfile_k8s [hunk 5]
```diff
@@ -6,36 +8,15 @@ parallel(
      buildDockerAndPublishImage('inbound-agent-maven:jdk8', [
       mainBranch: principalBranch,
```
Comment (lemeurherve) [inline Jenkinsfile_k8s:9]:
```suggestion
```

File: Jenkinsfile_k8s [hunk 6]
```diff
@@ -6,36 +8,15 @@ parallel(
...
-      mainBranch: principalBranch,
-      dockerfile: 'maven/jdk11/Dockerfile',
-    ])
-  },
-  'maven-jdk17': {
-     buildDockerAndPublishImage('inbound-agent-maven:jdk17', [
+  'maven-jdk8-nanoserver': {
+     buildDockerAndPublishImage('inbound-agent-maven:jdk8-nanoserver', [
+      dockerfile: 'maven/jdk8/Dockerfile.nanoserver',
       mainBranch: principalBranch,
```
Comment (lemeurherve) [inline Jenkinsfile_k8s:17]:
```suggestion
```

File: Jenkinsfile_k8s [hunk 7]
```diff
@@ -1,41 +1,20 @@
+@Library('pipeline-library@pull/372/head') _
```
Comment (lemeurherve) [inline Jenkinsfile_k8s:1]:
```suggestion
@Library('pipeline-library@pull/385/head') _
```

File: Jenkinsfile_k8s [hunk 8]
```diff
@@ -1,41 +1,20 @@
...
-    ])
-  },
-  'ruby': {
-     buildDockerAndPublishImage('inbound-agent-ruby', [
-      mainBranch: principalBranch,
-      dockerfile: 'ruby/Dockerfile',
+  'maven-jdk8-nanoserver': {
+     buildDockerAndPublishImage('inbound-agent-maven:jdk8-nanoserver', [
+      dockerfile: 'maven/jdk8/Dockerfile.nanoserver',
+      agentLabels: 'windows',
```
Comment (lemeurherve) [inline Jenkinsfile_k8s:16]:
```suggestion
      agentLabels: 'windows',
      platform: 'windows/amd64',
```

File: common-cst-windows.yml [hunk 9]
```diff
@@ -0,0 +1,31 @@
+schemaVersion: 2.0.0
+
+metadataTest:
+  entrypoint: ["pwsh.exe", "-f", "C:/ProgramData/Jenkins/jenkins-agent.ps1"]
+
+fileExistenceTests:
+
+  - name: 'Jenkins agent'
+    path: 'C:\ProgramData\Jenkins\jenkins-agent.ps1'
```
Comment (lemeurherve) [inline common-cst-windows.yml:9]:
```suggestion
    path: 'C:/ProgramData/Jenkins/jenkins-agent.ps1'
```

File: common-cst-windows.yml [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    shouldExist: true
+    isExecutableBy: 'any'
+  - name: 'Jenkins agent jar file'
+    path: 'C:\ProgramData\Jenkins\agent.jar'
```
Comment (lemeurherve) [inline common-cst-windows.yml:13]:
```suggestion
    path: 'C:/ProgramData/Jenkins/agent.jar'
```

File: common-cst-windows.yml [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    shouldExist: true
+  - name: "Default user's home"
+    path: 'C:\Jenkins'
```
Comment (lemeurherve) [inline common-cst-windows.yml:16]:
```suggestion
    path: 'C:/Jenkins'
```

File: common-cst-windows.yml [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+    shouldExist: true
+  - name: 'cmd'
+    path: 'C:\Windows\System32\cmd.exe'
```
Comment (lemeurherve) [inline common-cst-windows.yml:19]:
```suggestion
    path: 'C:/Windows/System32/cmd.exe'
```

File: common-cst-windows.yml [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+    shouldExist: true
+    isExecutableBy: 'any'
+  - name: 'Git'
+    path: 'C:\tools\git.exe'
```
Comment (lemeurherve) [inline common-cst-windows.yml:23]:
```suggestion
    path: 'C:/tools/git.exe'
```

File: common-cst-windows.yml [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+
+  - name: 'Jenkins agent'
+    path: 'C:/ProgramData/Jenkins/jenkins-agent.ps1'
+    shouldExist: true
+    isExecutableBy: 'any'
+  - name: 'Jenkins agent jar file'
+    path: 'C:/ProgramData/Jenkins/agent.jar'
+    shouldExist: true
+  - name: "Default user's home"
+    path: 'C:/Jenkins/'
```
Comment (lemeurherve) [inline common-cst-windows.yml:16]:
```suggestion
    path: 'C:/Jenkins'
```

File: common-cst-windows.yml [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+
+  - name: 'Jenkins agent'
+    path: 'C:/ProgramData/Jenkins/jenkins-agent.ps1'
+    shouldExist: true
+    isExecutableBy: 'any'
+  - name: 'Jenkins agent jar file'
+    path: 'C:/ProgramData/Jenkins/agent.jar'
+    shouldExist: true
+  - name: "Default user's home"
+    path: 'C:/Jenkins'
```
Comment (lemeurherve) [inline common-cst-windows.yml:16]:
```suggestion
    path: 'C:/users/jenkins/'
```

File: common-cst-windows.yml [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+    shouldExist: true
+  - name: 'cmd'
+    path: 'C:/Windows/System32/cmd.exe'
+    shouldExist: true
+    isExecutableBy: 'any'
+  - name: 'Git'
+    path: 'C:/tools/git.exe'
```
Comment (lemeurherve) [inline common-cst-windows.yml:23]:
```suggestion
    path: 'C:/mingit/mingw64/git.exe'
```

File: common-cst-windows.yml [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+    shouldExist: true
+  - name: "Default user's home"
+    path: 'C:/users/jenkins/'
+    shouldExist: true
+  - name: 'cmd'
+    path: 'C:/Windows/System32/cmd.exe'
+    shouldExist: true
+    isExecutableBy: 'any'
+  - name: 'Git'
+    path: 'C:/mingit/mingw64/git.exe'
```
Comment (lemeurherve) [inline common-cst-windows.yml:23]:
```suggestion
    path: 'C:/mingit/mingw64/bin/git.exe'
```

File: common-cst-windows.yml [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+  - name: 'Git'
+    path: 'C:/mingit/mingw64/bin/git.exe'
+    shouldExist: true
+    isExecutableBy: 'any'
+
+
+# fileContentTests:
+#   - name: 'Default user exists with the correct non-root UID/GID'
+#     path: '/etc/passwd'
+#     expectedContents: ['.*jenkins:x:100\d:100\d.*']
```
Comment (lemeurherve) [inline common-cst-windows.yml:31]:
```suggestion

```

File: maven/jdk11/Dockerfile.windowsservercore-1809 [hunk 19]
```diff
@@ -0,0 +1,2 @@
+# Use https://github.com/jenkinsci/docker-inbound-agent as base for now
+FROM jenkins/inbound-agent:windowsservercore-1809 AS inbound-windows-server-core
```
Comment (lemeurherve) [inline maven/jdk11/Dockerfile.windowsservercore-1809:2]:
```suggestion
FROM jenkins/inbound-agent:windowsservercore-1809 AS inbound-windows-server-core

ENTRYPOINT [pwsh.exe -f C:/ProgramData/Jenkins/jenkins-agent.ps1]

```

File: maven/jdk11/Dockerfile.windowsservercore-1809 [hunk 20]
```diff
@@ -0,0 +1,5 @@
...
+
+ENTRYPOINT [pwsh.exe -f C:/ProgramData/Jenkins/jenkins-agent.ps1]
+
```
Comment (lemeurherve) [inline maven/jdk11/Dockerfile.windowsservercore-1809:5]:
```suggestion
ENTRYPOINT [pwsh.exe -f C:/ProgramData/Jenkins/jenkins-agent.ps1]
```

Comment (lemeurherve) [inline maven/jdk11/Dockerfile.windowsservercore-1809:5]:
```suggestion
ENTRYPOINT [pwsh.exe -f C:/ProgramData/Jenkins/jenkins-agent.ps1]
```

**Review by dduportal — APPROVED** [review]
🎉

**Pull request merged**